File: day1/Day1.java
import java.io.*;
import java.util.*;
public class Day1 {

   public static void main(String[] args) throws FileNotFoundException {
      Scanner input = new Scanner(new File("input.txt"));
      String in = input.next();
      a(in);
      b(in);
   }

   public static void a(String in) {
      int sum = 0;
      for(int i = 0; i < in.length(); i++) {
         if(in.charAt(i) == in.charAt((i + 1)%in.length())) {
            sum += Integer.parseInt(in.substring(i, i+1));
         }
      }
      System.out.println(sum);
   }

   public static void b(String in) {
      int sum = 0;
      for(int i = 0; i < in.length(); i++) {
         if(in.charAt(i) == in.charAt((i + in.length()/2)%in.length())) {
            sum += Integer.parseInt(in.substring(i, i+1));
         }
      }
      System.out.println(sum);
   }

}

File: day1/Day1.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

int main (int argc, char ** argv) {
   ifstream inFile;
   inFile.open("input.txt");
   std::string s;
   inFile >> s;
   int sum = 0;
   for(int i = 0; i < s.size(); i++) {
      if(s[i] == s[(i+s.size()/2)%s.size()]) {
         sum += s[i] - '0';
      }
   }
   cout << sum << endl;
}